### PR TITLE
[VPU] use-heap-after-free-fix

### PR DESCRIPTION
**Pull request by KorA6**
### Details:
- #8466
- #8509

### Tickets:
 - *69790*

### CODE REVIEW

File: inference-engine/thirdparty/movidius/mvnc/src/mvnc_api.c [hunk 1]
```diff
@@ -3146,41 +3139,36 @@ ncStatus_t ncFifoDestroy(struct ncFifoHandle_t ** fifoHandle)
...
         CHECK_MUTEX_SUCCESS(pthread_mutex_unlock(&d->graph_stream_m));
         mvLog(MVLOG_WARN, "myriad NACK\n");
-        return NC_ERROR;
+        rc = NC_ERROR;
     }
     CHECK_MUTEX_SUCCESS(pthread_mutex_unlock(&d->graph_stream_m));
 
     CHECK_MUTEX_SUCCESS(pthread_mutex_lock(&d->dev_data_m));
     if (deallocateFifo(handle)) {
         CHECK_MUTEX_SUCCESS(pthread_mutex_unlock(&d->dev_data_m));
```
Comment (DariaMityagina) [inline inference-engine/thirdparty/movidius/mvnc/src/mvnc_api.c:3166]: Why not deleted like [here](https://github.com/openvinotoolkit/openvino/pull/8509/files#:~:text=mvLog(MVLOG_WARN%2C%20%22failed%20deallocateFifo%5Cn%22)%3B) ?

Comment (DariaMityagina) [reply inference-engine/thirdparty/movidius/mvnc/src/mvnc_api.c:3166]: And in some other places.

File: inference-engine/thirdparty/movidius/mvnc/src/mvnc_api.c [hunk 2]
```diff
@@ -3146,41 +3139,36 @@ ncStatus_t ncFifoDestroy(struct ncFifoHandle_t ** fifoHandle)
...
+        rc =  NC_INVALID_PARAMETERS;
     }
     CHECK_MUTEX_SUCCESS(pthread_mutex_unlock(&d->dev_data_m));
 
-    free(fh->private_data);
-    fh->private_data = NULL;
-    free(fh);
-    *fifoHandle = NULL;
+    FREE_AND_NULL(fh->private_data, fh->private_data, fh, *fifoHandle);
     return NC_OK;
```
[the remark this answers is not in the document]
Comment (KorA6) [reply inference-engine/thirdparty/movidius/mvnc/src/mvnc_api.c]: Oups. My bad.
Changed.

File: inference-engine/thirdparty/movidius/mvnc/src/mvnc_api.c (hunk 1 above)
Comment (KorA6) [reply inference-engine/thirdparty/movidius/mvnc/src/mvnc_api.c]: I agree with you. Changed

File: inference-engine/thirdparty/movidius/mvnc/src/mvnc_api.c [hunk 3]
```diff
@@ -3151,25 +3151,22 @@ ncStatus_t ncFifoDestroy(struct ncFifoHandle_t ** fifoHandle)
...
     } else if (checkGraphMonitorResponse(d->graph_monitor_stream_id)) {
-        CHECK_MUTEX_SUCCESS(pthread_mutex_unlock(&d->graph_stream_m));
         mvLog(MVLOG_WARN, "myriad NACK\n");
         rc = NC_ERROR;
     }
     CHECK_MUTEX_SUCCESS(pthread_mutex_unlock(&d->graph_stream_m));
 
     CHECK_MUTEX_SUCCESS(pthread_mutex_lock(&d->dev_data_m));
     if (deallocateFifo(handle)) {
-        CHECK_MUTEX_SUCCESS(pthread_mutex_unlock(&d->dev_data_m));
```
Comment (DariaMityagina) [inline inference-engine/thirdparty/movidius/mvnc/src/mvnc_api.c:3175]: I think the warning will not be superfluous:)

Comment (KorA6) [reply inference-engine/thirdparty/movidius/mvnc/src/mvnc_api.c:3175]: Added

**Review by DariaMityagina — APPROVED** [review]

**Review by AntonDudchenko — APPROVED** [review]

**Pull request merged**